crates/stdx/Cargo.toml

```diff
@@ -15,6 +15,7 @@ doctest = false
 libc = "0.2.135"
 backtrace = { version = "0.3.65", optional = true }
 always-assert = { version = "0.1.2", features = ["log"] }
+nohash-hasher.workspace = true
```

Comment:
let's actually kick this out from stdx, it doesn't really have a place in here (the original functionality was put here because it didnt really deserve a separate crate, but thats no longer the case since we are depending on this now) So instead let's add this dep where needed